client/profile-wizard/steps/store-details/index.js

```diff
@@ -393,8 +437,36 @@ export default compose(
 			isOnboardingRequesting( 'updateProfileItems' ) ||
 			isUpdateSettingsRequesting( 'general' ) ||
 			isResolving( 'getOption', [ 'woocommerce_allow_tracking' ] );
+		const isLoading = ! hasFinishedResolutionOnboarding(
+			'getProfileItems'
+		);
+		// Check if a store address is set so that we don't default
+		// to WooCommerce's default country of the UK.
+		const countryState =
+			( settings.woocommerce_store_address &&
+				settings.woocommerce_default_country ) ||
+			'';
+
+		const initialValues = {
+			addressLine1: settings.woocommerce_store_address || '',
+			addressLine2: settings.woocommerce_store_address_2 || '',
+			city: settings.woocommerce_store_city || '',
+			countryState,
+			postCode: settings.woocommerce_store_postcode || '',
+			isClient: profileItems.setup_client || false,
+			isAgreeMarketing:
+				typeof profileItems.is_agree_marketing === 'boolean'
+					? profileItems.is_agree_marketing
+					: true,
+			storeEmail:
+				typeof profileItems.store_email === 'string'
+					? profileItems.store_email
+					: '',
```

Comment:
The default email is planned to be pre-filled with Jetpack, or if not available, the WordPress's admin email instead. Follow-up incoming for this soon.